### PR TITLE
[NEW] Room swipe actions: mark as read/unread, hide, fav

### PR DESCRIPTION
**Pull request by pranavpandey1998official**
Added Feture to mark read and unread , hide channel,group , dm, and 
feture to fav and unfav 
ezgif com-video-to-gif

### CODE REVIEW

**Comment by geekgonecrazy** [comment]
:heart: One of the last major features that keeps me from being able to commit to using this app for extended period :)

**Comment by diegolmello** [comment]
@pranavpandey1998official following our offline talk, pick https://github.com/RocketChat/Rocket.Chat.ReactNative/commit/be717ca3db32a94e3d4c9fb0ba319aae72908282

File: app/presentation/RoomItem/index.js [hunk 1]
```diff
@@ -63,10 +72,92 @@ export default class RoomItem extends React.Component {
...
+	}
+
+	renderLeftActions = (progress, dragX) => {
+		const { isRead } = this.props;
+		const trans = dragX.interpolate({
+			inputRange: [0, 80, 81],
+			outputRange: [0, 0, 1]
+		});
+		return (
+			<RectButton style={[styles.action, { backgroundColor: '#497AFC' }]}>
```
Comment (diegolmello) [inline app/presentation/RoomItem/index.js]: Inline style

File: app/presentation/RoomItem/index.js [hunk 2]
```diff
@@ -63,10 +72,92 @@ export default class RoomItem extends React.Component {
...
+						</View>
+					)}
+				</Animated.View>
+			</RectButton>
+		);
+	};
+
+	renderRightActions = (progress, dragX) => {
+		const { favorite } = this.props;
+		const { width } = Dimensions.get('window');
```
Comment (diegolmello) [inline app/presentation/RoomItem/index.js]: Use `react-native-responsive-ui`

File: app/presentation/RoomItem/index.js [hunk 3]
```diff
@@ -97,30 +184,43 @@ export default class RoomItem extends React.Component {
 		}
 
 		return (
-			<RectButton
-				onPress={onPress}
-				activeOpacity={0.8}
-				underlayColor='#e1e5e8'
-				testID={testID}
+			<Swipeable
+				ref={(ref) => { this.swipeableRow = ref; }}
```
Comment (diegolmello) [inline app/presentation/RoomItem/index.js]: Use method

File: package.json [hunk 4]
```diff
@@ -130,7 +133,10 @@
     },
     "transform": {
       "^.+\\.js$": "<rootDir>/node_modules/react-native/jest/preprocessor.js"
-    }
+	},
+	"transformIgnorePatterns": [
+		"node_modules/!redux-enhancer-react-native-appstate"
```
Comment (diegolmello) [inline package.json]: Is this still necessary?

File: package.json [hunk 5]
```diff
@@ -120,7 +120,10 @@
     "testPathIgnorePatterns": [
       "e2e",
       "node_modules"
-    ],
+	],
+	"setupFiles": [
+		"./node_modules/react-native-gesture-handler/jestSetup.js"
```
Comment (diegolmello) [inline package.json]: Is this still necessary?

File: package.json (hunk 4 above)
Comment (pranavpandey1998official) [reply package.json]: removed

File: package.json (hunk 5 above)
Comment (pranavpandey1998official) [reply package.json]: removed

File: app/presentation/RoomItem/index.js (hunk 2 above)
Comment (pranavpandey1998official) [reply app/presentation/RoomItem/index.js]: our pakage is v1.1.1 and useDimention feture is in 2+

File: app/presentation/RoomItem/index.js [hunk 6]
```diff
@@ -63,10 +93,243 @@ export default class RoomItem extends React.Component {
...
+				]}
+			>
+				<RectButton style={[styles.actionButtonLeft]} onPress={this.handleLeftButtonPress}>
+					<Animated.View
+						style={{ transform: [{ translateX: iconTrans }] }}
+					>
+						{isRead ? (
+							<View style={styles.actionView}>
+								<CustomIcon size={15} name='flag' color='white' />
+								<Text style={styles.actionText}>Unread</Text>
```
Comment (diegolmello) [inline app/presentation/RoomItem/index.js]: i18n

File: app/presentation/RoomItem/index.js [hunk 7]
```diff
@@ -63,10 +93,243 @@ export default class RoomItem extends React.Component {
...
+							</View>
+						) : (
+							<View style={styles.actionView}>
+								<CustomIcon size={15} name='check' color='white' />
+								<Text style={styles.actionText}>Read</Text>
```
Comment (diegolmello) [inline app/presentation/RoomItem/index.js]: i18n

File: app/presentation/RoomItem/index.js [hunk 8]
```diff
@@ -63,10 +93,243 @@ export default class RoomItem extends React.Component {
...
+					style={[
+						{ width: iconFavWidth },
+						{ transform: [{ translateX: iconFavTrans }] }
+					]}
+				>
+					<RectButton style={[styles.actionButtonRightFav]} onPress={this.toggleFav}>
+						{favorite ? (
+							<View style={styles.actionView}>
+								<CustomIcon size={17} name='Star-filled' color='white' />
+								<Text style={styles.actionText}>Unfavorite</Text>
```
Comment (diegolmello) [inline app/presentation/RoomItem/index.js]: i18n

File: app/presentation/RoomItem/index.js [hunk 9]
```diff
@@ -63,10 +93,243 @@ export default class RoomItem extends React.Component {
...
+							</View>
+						) : (
+							<View style={styles.actionView}>
+								<CustomIcon size={17} name='star' color='white' />
+								<Text style={styles.actionText}>Favorite</Text>
```
Comment (diegolmello) [inline app/presentation/RoomItem/index.js]: i18n

File: app/presentation/RoomItem/index.js [hunk 10]
```diff
@@ -63,10 +93,243 @@ export default class RoomItem extends React.Component {
...
+					{ transform: [{ translateX: iconHideTrans }] }
+				]}
+				>
+					<RectButton
+						style={[styles.actionButtonRightHide]}
+						onPress={this.handleHideButtonPress}
+					>
+						<View style={styles.actionView}>
+							<CustomIcon size={15} name='eye-off' color='white' />
+							<Text style={styles.actionText}>Hide</Text>
```
Comment (diegolmello) [inline app/presentation/RoomItem/index.js]: i18n

File: app/views/RoomsListView/index.js [hunk 11]
```diff
@@ -416,19 +440,24 @@ export default class RoomsListView extends React.Component {
 					alert={item.alert}
 					unread={item.unread}
 					userMentions={item.userMentions}
+					isRead={!((item.archived !== true && item.open === true) && (item.unread > 0 || item.alert === true))}
```
Comment (diegolmello) [inline app/views/RoomsListView/index.js]: Create a method for that.
Split conditions and avoid double negation.
Add comments.
```jsx
get isRead() {
  ...
}

...

isRead={this.isRead}
```

File: app/presentation/RoomItem/index.js [hunk 12]
```diff
@@ -63,10 +93,243 @@ export default class RoomItem extends React.Component {
...
+			return rightOffset;
+		}
+		return 0;
+	};
+
+	_handleRelease = (nativeEvent) => {
+		const { translationX } = nativeEvent;
+		const { rowState } = this.state;
+		let toValue = 0;
+		if (rowState === 0) { // if no option is opned
```
Comment (diegolmello) [inline app/presentation/RoomItem/index.js]: typo

File: app/presentation/RoomItem/index.js [hunk 13]
```diff
@@ -50,10 +62,28 @@ export default class RoomItem extends React.Component {
...
+		const dragX = new Animated.Value(0);
+		const rowOffSet = new Animated.Value(0);
+		this.rowTranslation = Animated.add(
+			rowOffSet,
+			dragX
+		);
+		this.state = {
+			dragX,
+			rowOffSet,
+			rowState: 0, // here 0 means closed 1 means right open and -1 left open
```
Comment (diegolmello) [inline app/presentation/RoomItem/index.js]: Simplify this comment.
```
// 0: closed, 1: right opened, -1: left opened
```

File: app/presentation/RoomItem/index.js [hunk 14]
```diff
@@ -63,10 +93,243 @@ export default class RoomItem extends React.Component {
...
+			outputRange: [0, 0, -(OPTION_WIDTH + 10), 0, 0]
+		});
+		return (
+			<Animated.View
+				style={[
+					styles.leftAction,
+					{ transform: [{ translateX: trans }] }
+				]}
+			>
+				<RectButton style={[styles.actionButtonLeft]} onPress={this.handleLeftButtonPress}>
```
Comment (diegolmello) [inline app/presentation/RoomItem/index.js]: Unnecessary array

File: app/presentation/RoomItem/index.js [hunk 15]
```diff
@@ -63,10 +93,243 @@ export default class RoomItem extends React.Component {
...
+	renderRightActions = () => {
+		const { favorite } = this.props;
+		const { width } = Dimensions.get('window');
+		const trans = this.rowTranslation.interpolate({
+			inputRange: [-OPTION_WIDTH, 0],
+			outputRange: [width - OPTION_WIDTH, width]
+		});
+		const iconFavTrans = this.rowTranslation.interpolate({
+			inputRange: [-2 * OPTION_WIDTH, 0],
+			outputRange: [0, 0]
```
Comment (diegolmello) [inline app/presentation/RoomItem/index.js]: Is this animation necessary? 🤔

File: app/presentation/RoomItem/index.js [hunk 16]
```diff
@@ -63,10 +93,243 @@ export default class RoomItem extends React.Component {
...
+		this._animateRow(toValue);
+	}
+
+	_animateRow = (toValue) => {
+		const { dragX, rowOffSet } = this.state;
+		rowOffSet.setValue(this._value);
+		dragX.setValue(0);
+		Animated.spring(rowOffSet, {
+			toValue,
+			bounciness: 0
```
Comment (diegolmello) [inline app/presentation/RoomItem/index.js]: `bounciness: 10`

File: app/views/RoomsListView/index.js (hunk 11 above)
Comment (pranavpandey1998official) [reply app/views/RoomsListView/index.js]: I cannot have a get as the item I am getting is coming from FlatList I have made a simple function ,

File: app/presentation/RoomItem/index.js (hunk 16 above)
Comment (pranavpandey1998official) [reply app/presentation/RoomItem/index.js]: are you sure it doesn't look good

File: app/presentation/RoomItem/index.js (hunk 15 above)
Comment (pranavpandey1998official) [reply app/presentation/RoomItem/index.js]: removed

File: app/presentation/RoomItem/index.js (hunk 16 above)
Comment (pranavpandey1998official) [reply app/presentation/RoomItem/index.js]: I am adding 5

**Pull request merged**